daemon.c

```diff
@@ -1540,8 +1556,10 @@ void *receiveOOK(void *param) {
 					/* Let's do a little filtering here as well */
 					if(r.length >= hw->minrawlen && r.length <= hw->maxrawlen) {
 						receive_queue(r.pulses, r.length, plslen, hw->hwtype);
+						r.length = 0;
 					}
-					r.length = 0;
+					// Avoid that short footer pulses (X10, IMPULS, ...) interfere with long header /sync pulses
+					if(r.length>20)r.length = 0;	// Avoid a buffer reset and preserve potential Header / SYNC / Start pulses
```

Comment:
The S*mfy protocol has pulse durations in its header that do exceed the duration of the footer pulses for X10 and IMPULS.

Thus those pulses do split the buffer (or to be more accurat, they do reset the buffer length pointer), causing loss of pulses required, as they are needed for synchronization of the bi-phase-manchester decoder logic.

 

There are protocols waiting to be merged (most notable OREGON V2.1) having no footer pulses, but header pulses instead, for those protocols the same does apply.

 

In the past this problem was neglected, the first pulsetrain was discarded, the header from the second (third, forth, …) pulsetrain was interpreted as footer, and the last pulsetrain was discarded.

 

This won’t work for OREGON, as OREGON is using  a pulsed sequence as footer criteria to split pulsetrains and one header pulse only.

 

There is no compatibility problem with existing protocols, as that check is limited to the first 20 pulses, but existing protocols are 50 pulses minimum.

 

 

Von: CurlyMoo [mailto:notifications@github.com] 
Gesendet: Sonntag, 20. Dezember 2015 18:11
An: pilight/pilight
Cc: wo-rasp
Betreff: Re: [pilight] S*mfy Protocol (#200)

 

In daemon.c <https://github.com/pilight/pilight/pull/200#discussion_r48105670> :

>                                     }
> -                                   r.length = 0;
> +                                   // Avoid that short footer pulses (X10, IMPULS, ...) interfere with long header /sync pulses
> +                                   if(r.length>20)r.length = 0;   // Avoid a buffer reset and preserve potential Header / SYNC / Start pulses

I don't think i understand  :smile: <https://assets-cdn.github.com/images/icons/emoji/unicode/1f604.png> 

—
Reply to this email directly or view it on GitHub <https://github.com/pilight/pilight/pull/200/files#r48105670> .  <https://github.com/notifications/beacon/AG9MV1YNUpD5Xv7Dyve_QpU4-1_KWiJZks5pRtiggaJpZM4GUJeH.gif>